### PR TITLE
MTS-434: Metadata cache doesn't work properly on Salesforce Query Operation

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

**Comment by lucianoRM** [comment]
--validate

File: tests/test-extensions/metadata-extension/src/main/java/org/mule/test/metadata/extension/MetadataOperations.java [hunk 1]
```diff
@@ -276,6 +278,15 @@ public String returnQuery(@MetadataKeyId String query) {
     return query;
   }
 
+  @Query(translator = MetadataExtensionQueryTranslator.class,
+      entityResolver = MetadataExtensionEntityResolver.class,
+      nativeOutputResolver = NativeQueryOutputResolver.class)
+  @MediaType(TEXT_PLAIN)
+  public String queryWithOtherMetadata(@MetadataKeyId(QueryExtraResolver.class) String query,
```
Comment (ndinu) [inline tests/test-extensions/metadata-extension/src/main/java/org/mule/test/metadata/extension/MetadataOperations.java:285]: I think that if you use DSQL, you cannot have input resolvers nor metadata key resolvers. You are adding the to the TypeResolversModelProperty, but they are not in the ResolversFactory.

**Comment by lucianoRM** [comment]
--validate

**Comment by lucianoRM** [comment]
--validate

**Comment by lucianoRM** [comment]
--validate

**Pull request merged**